### ISSUE TITLE
NewPipe REST APIs

### ISSUE THREAD

**Issue by yausername** [TeamNewPipe/NewPipeExtractor#172]
Hi,
I have tried to create REST APIs for NewPipeExtractor.
This could be used for porting NewPipe to other platforms as discussed [here](https://github.com/TeamNewPipe/NewPipe/issues/1051).
The current implementation is just returning the POJOs as json.

It would be interesting to get your feedback.
@theScrabi @TobiGr @TheAssassin @mauriciocolli @PrestonN @omarroth

Code - https://github.com/yausername/NewPipeREST
Swagger UI (Documentation) - https://app.swaggerhub.com/apis-docs/yayausername/new-pipe_rest/1.0.0
Test Server - https://newpipe-rest.herokuapp.com/api/v1
OpenAPI spec - https://github.com/yausername/NewPipeREST/blob/master/docs/openapi.yaml

**Comment by TheAssassin** [comment]
Interesting approach. That means you still need some extra interface, but at least NPE can be re-used.

**Comment by yausername** [comment]
> least

Yes, I am thinking of creating an npm module that would wrap this server. It could then be used in electron based apps.

**Comment by PrestonN** [comment]
Very interesting.  I'll try to find some time this week to dig into this and poke around.  At a glance everything seems nice and well organized.  I'll come back and leave my feedback once I've had a chance to play around with everything.

**Comment by theScrabi** [comment]
You are going to make @omarroth concurrence.
